api/service/block.ts

```diff
@@ -0,0 +1,112 @@
+import { Block } from 'shared/model/block';
+import { BlockRepo } from 'api/repository/block';
+import Response from 'shared/model/response';
+const OrderBy = require('../../helpers/orderBy.js');
+
+interface ILoadBlockParams {
+    limit: number;
+    lastId: string;
+}
+
+interface IListParams {
+    generatorPublicKey?: string;
+    numberOfTransactions?: number;
+    previousBlock?: string;
+    height?: number;
+    totalAmount?: number;
+    totalFee?: number;
+    reward?: number;
+    limit?: number;
+    offset?: number;
+    orderBy?: string;
+}
+
+export class BlockService {
+    private blockRepo = new BlockRepo();
+    private lastBlock: Block;
+    private sortFields = [
+        'id',
+        'timestamp',
+        'height'
+    ];
+
+    public loadBlocksData(params: ILoadBlockParams) : Block[] {
+        return [];
+    }
+
+    public getLastBlock(): Block {
+        // load block from db?
+        return this.lastBlock;
+    }
+
+    /**
+     * @deprecated
+     */
+    public setLastBlock(block: Block): Block {
+        this.lastBlock = block;
+        return this.lastBlock;
+    }
+
+    public dbRead(raw: {[key: string]: any}, radix: number = 10): Block { return undefined; }
+
+    public calculateFee(): number { return undefined; }
+
+    public list(filter: IListParams): Response<{ blocks: Block[], count: number }> {
+        const params: IListParams = {};
+        const where: string[] = [];
+
+        if (filter.generatorPublicKey) {
+            where.push('"b_generatorPublicKey"::bytea = ${generatorPublicKey}');
+            params.generatorPublicKey = filter.generatorPublicKey;
+        }
+
+        if (filter.previousBlock) {
+            where.push('"b_previousBlock" = ${previousBlock}');
+            params.previousBlock = filter.previousBlock;
+        }
+
+        if (filter.height === 0 || filter.height > 0) {
```

Comment:
`filter.height === 0` can block height be zero?

Reply:
I hope it can't, fixing